components/renku_data_services/notifications/db.py

```diff
@@ -0,0 +1,98 @@
+"""Adapters for notification database classes."""
+
+from collections.abc import Callable
+from datetime import UTC, datetime
+
+from sqlalchemy import select
+from sqlalchemy.ext.asyncio import AsyncSession
+from ulid import ULID
+
+from renku_data_services import base_models, errors
+from renku_data_services.notifications import models
+from renku_data_services.notifications import orm as schemas
+
+
+class NotificationsRepository:
+    """Repository for Notifications."""
+
+    def __init__(
+        self,
+        session_maker: Callable[..., AsyncSession],
+    ):
+        self.session_maker = session_maker
+
+    async def create_alert(self, user: base_models.APIUser, alert: models.UnsavedAlert) -> models.Alert:
+        """Insert a new alert into the database."""
+        if user.id is None:
+            raise errors.UnauthorizedError(message="You do not have the required permissions for this operation.")
+        if not user.is_admin:
+            raise errors.ForbiddenError(message="You do not have the required permissions for this operation.")
+
+        async with self.session_maker() as session:
+            query = (
+                select(schemas.AlertORM)
+                .where(schemas.AlertORM.user_id == alert.user_id)
+                .where(schemas.AlertORM.title == alert.title)
+                .where(schemas.AlertORM.message == alert.message)
+                .where(schemas.AlertORM.session_name == alert.session_name)
+                .where(schemas.AlertORM.resolved_at.is_(None))
```

Comment:
Discussion: with this logic, if I post an alert with just a slightly different message, then I can add as many alerts on the same session as I want.

To avoid this, I think it could be useful to have a field called `event_type` which is used to specify the type of alert, e.g. `session_oom` or `session_memory_limit`, etc. Then we can de-duplicate based on `user_id, event_type, session_name`.